version: '3.8'

services:
  django:
    build:
      # . mean inside current directory
      context: .
      dockerfile: Dockerfile.dev

    networks:
      - main

    command:
      - gunicorn
      - crypto_ext_backend.wsgi:application
      - --bind
      - 0.0.0.0:8000
#      - daphne -p 8001 crypto_ext_backend.asgi:application
    depends_on:
      - redis
      - rabbitmq

  channel:
    image: django
    build:
      context: .
      dockerfile: Dockerfile.dev

    networks:
      - main

    depends_on:
      - redis
      - rabbitmq
      - django
    command: "daphne crypto_ext_backend.asgi:application --bind 0.0.0.0 --port 9000"

  celery_worker:
    image: django
    build:
      context: .
      dockerfile: Dockerfile.dev
    command: sh -c "celery -A crypto_ext_backend worker -l info"
    networks:
      - main
    depends_on:
      - django

  celery_beat:
    image: django
    build:
      context: .
      dockerfile: Dockerfile.dev
    command:  sh -c "celery -A crypto_ext_backend beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler"
    networks:
      - main
    depends_on:
      - celery_worker

  nginx:
    build:
      context: ./nginx

    depends_on:
      - django
    networks:
      - main
    restart: on-failure
    ports:
      - "8000:80"
  redis:
    image: redis:latest
    restart: on-failure
    networks:
      - main
    ports:
      - "6378:6379"

  rabbitmq:
    container_name: rabbitmq
    hostname: rabbitmq
    image: rabbitmq:latest
    networks:
      - main
    ports:
      - "5672:5672"
    restart: on-failure


networks:
  main:


